custom/icds_reports/utils/aggregation_helpers/mbt.py

```diff
@@ -143,81 +153,114 @@ def columns(self):
+                't.bf_at_birth_born_in_month',
+                't.ebf_eligible',
+                't.ebf_in_month',
+                't.ebf_not_breastfeeding_reason',
+                't.ebf_drinking_liquid',
+                't.ebf_eating',
+                't.ebf_no_bf_no_milk',
+                't.ebf_no_bf_pregnant_again',
+                't.ebf_no_bf_child_too_old',
+                't.ebf_no_bf_mother_sick',
+                't.cf_eligible',
+                't.cf_in_month',
+                't.cf_diet_diversity',
+                't.cf_diet_quantity',
+                't.cf_handwashing',
+                't.cf_demo',
+                't.fully_immunized_eligible',
+                't.fully_immunized_on_time',
+                't.fully_immunized_late',
+                't.counsel_ebf',
+                't.counsel_adequate_bf',
+                't.counsel_pediatric_ifa',
+                't.counsel_comp_feeding_vid',
+                't.counsel_increase_food_bf',
+                't.counsel_manage_breast_problems',
+                't.counsel_skin_to_skin',
+                't.counsel_immediate_breastfeeding',
+                't.recorded_weight',
+                't.recorded_height',
+                't.has_aadhar_id',
+                't.thr_eligible',
+                't.pnc_eligible',
+                't.cf_initiation_in_month',
+                't.cf_initiation_eligible',
+                't.height_measured_in_month',
+                't.current_month_stunting',
+                't.stunting_last_recorded',
+                't.wasting_last_recorded',
+                't.current_month_wasting',
+                't.valid_in_month',
+                't.valid_all_registered_in_month',
+                't.ebf_no_info_recorded',
+                't.dob',
+                't.sex',
+                't.age_tranche',
+                't.caste',
+                't.disabled',
+                't.minority',
+                't.resident',
+                't.immunization_in_month',
+                't.days_ration_given_child',
+                't.zscore_grading_hfa',
+                't.zscore_grading_hfa_recorded_in_month',
+                't.zscore_grading_wfh',
+                't.zscore_grading_wfh_recorded_in_month',
+                't.muac_grading',
+                'ccs.case_id as ccs_record_case_id',
+                't.date_death')
+
+    def query(self):
+        return """
+        COPY (SELECT {columns} FROM {table} t
+        LEFT JOIN awc_location awc on t.awc_id=awc.doc_id
+        LEFT JOIN "{person_cases_ucr}" mother on mother.doc_id=t.mother_case_id
+          AND awc.state_id = mother.state_id
+          AND lower(substring(mother.state_id, '.{{3}}$'::text)) = '{state_id_last_3}'
+        LEFT JOIN "{ccs_cases_monthly_ucr}" ccs on ccs.person_case_id=mother.doc_id AND ccs.add=t.dob
+        WHERE awc.state_id='{state_id}' AND t.month='{month}')
+        TO '{output}' WITH CSV HEADER;
```

Comment:
> COPY with a file name instructs the PostgreSQL server to directly read from or write to a file. The file must be accessible by the PostgreSQL user (the user ID the server runs as) and the name must be specified from the viewpoint of the server. When PROGRAM is specified, the server executes the given command and reads from the standard output of the program, or writes to the standard input of the program. The command must be specified from the viewpoint of the server, and be executable by the PostgreSQL user. When STDIN or STDOUT is specified, data is transmitted via the connection between the client and the server.

https://www.postgresql.org/docs/9.6/sql-copy.html

I'm skeptical that this is working how you think it is. The doc makes me think that this is going to write to /tmp on the server which I think is not what you want.